### PR TITLE
Restricitve Boltzmann Machines Implementation

### PR DESCRIPTION
**Pull request by maheswarantp**
## Description
The notebooks provide a JAX implementation of Restrictive Boltzmann Machines, including their training.
It utilizes automatic differentiation capability provided by JAX to compute the gradients. It has been tested on Colab GPU as well as CPU

Reconstruction images are attached below

### Colab link
[rbm_training,ipynb](https://colab.research.google.com/drive/1VG1Aw7EwlHvL5h1M8JJFzQtg9UJXzPgX#scrollTo=a70a65c7)
[sigmoid_rbm_training.ipynb](https://colab.research.google.com/drive/1-BtK72Wpvn1u2tY4qvS1ytWRTVPK8s0n#scrollTo=5c3a0e89)

### Issue 
[ make pedagogical JAX EBM demos #674 ](https://github.com/probml/pyprobml/issues/674)

### Figures
[figure](https://gist.github.com/maheswarantp/59d4d63ea3e9e6ba153402833b13efe9)

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

### CODE REVIEW

File: notebooks/rbm_training.ipynb [hunk 1]
```diff
@@ -0,0 +1,638 @@
+{
```
Comment (gerdm) [inline notebooks/rbm_training.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #21.        for i in range(steps):
</pre><p>Consider replacing for <a href="https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.scan.html" target="_blank">jax.lax.scan</a> or the <a href="https://jax.readthedocs.io/en/latest/jax.experimental.loops.html" target="_blank">loops module</a> in Jax</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/probml-notebooks/pull/91/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

**Comment by murphyk** [comment]
I've already added https://github.com/probml/probml-notebooks/blob/main/notebooks/rbm_contrastive_divergence.ipynb, so don't need another demo. But thanks anyway!
